tests/sdk/v2/conversions/test_yaml_exporter.py

```diff
@@ -18,950 +18,967 @@ def _model_dict(model_obj):
     return json.loads(model_obj.json(exclude_none=True))
```

Comment:
the changes here seem unrelated to the main focus of the PR (at least I'm finding it hard to see a link).
wdyt about making this a separate commit/PR after it lands in `breaking-changes`?

Reply:
The changes in this file are caused by a combination of factors:
1. I've made `WorkflowDef` raise a warning when the dumped SDK version is different than the parse-time SDK version.
2. We treat warnings as errors in our test suite.
3. Pytest lets silencing warnings on a per test-case basis, but not on per-statement. [Pytest docs](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html) mention `with warnings.catch_warnings(): warnings.simplefilter("ignore")` but I couldn't make it work with our stuff. Probably (2) captures warnings as errors regardless of silencing them.
4. Workflow defs in this file were hardcoded as module constants evaluated at module load time instead of fixtures evaluated at test execution time. I turned them into thunks.

I can see how these changes make this PR difficult to read. I'll be happy to extract these changes into a separate PR.

Reply:
Split into https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/107